File: result/metagpt/claude/humaneval/119.c
#include <stdio.h>
#include <string.h>

char* match_parens(char* lst[2]) {
    char l1[1000]; // Assuming a maximum length for the concatenated string
    int i, count = 0;
    int can = 1;

    // Concatenate the two strings
    strcpy(l1, lst[0]);
    strcat(l1, lst[1]);

    // Check if the concatenated string is balanced
    for (i = 0; i < strlen(l1); i++) {
        if (l1[i] == '(') count += 1;
        if (l1[i] == ')') count -= 1;
        if (count < 0) can = 0;
    }

    if (count != 0) return "No";
    if (can == 1) return "Yes";

    // Try the other order
    strcpy(l1, lst[1]);
    strcat(l1, lst[0]);

    count = 0;
    can = 1;

    for (i = 0; i < strlen(l1); i++) {
        if (l1[i] == '(') count += 1;
        if (l1[i] == ')') count -= 1;
        if (count < 0) can = 0;
    }

    if (can == 1) return "Yes";
    return "No";
}

File: result/metagpt/gpt4o/humaneval/119.rs
fn match_parens(lst: [&str; 2]) -> &'static str {
    let mut l1 = String::new();
    l1.push_str(lst[0]);
    l1.push_str(lst[1]);

    let mut count = 0;
    let mut can = true;

    for ch in l1.chars() {
        if ch == '(' {
            count += 1;
        } else if ch == ')' {
            count -= 1;
        }
        if count < 0 {
            can = false;
        }
    }

    if count != 0 {
        return "No";
    }
    if can {
        return "Yes";
    }

    l1.clear();
    l1.push_str(lst[1]);
    l1.push_str(lst[0]);

    count = 0;
    can = true;

    for ch in l1.chars() {
        if ch == '(' {
            count += 1;
        } else if ch == ')' {
            count -= 1;
        }
        if count < 0 {
            can = false;
        }
    }

    if can {
        "Yes"
    } else {
        "No"
    }
}

fn main() {
    let lst = ["(()", ")()"];
    let result = match_parens(lst);
    println!("Balanced: {}", result);
}